### PR TITLE
Fixed #34821 -- Prevented DEFAULT_FILE_STORAGE/STATICFILES_STORAGE settings from mutating the main STORAGES.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-34821, check out [logs](https://djangoci.com/job/main-random/database=mysql,label=focal,python=python3.11/).

Regression in 6b965c600054f970bdf94017ecf2e0e6e0a4326b.

### CODE REVIEW

File: django/conf/__init__.py [hunk 1]
```diff
@@ -222,19 +222,29 @@ def __init__(self, settings_module):
...
-                "BACKEND": self.DEFAULT_FILE_STORAGE
-            }
+            setattr(
+                self,
+                "STORAGES",
+                {
+                    **self.STORAGES,
+                    DEFAULT_STORAGE_ALIAS: {"BACKEND": self.DEFAULT_FILE_STORAGE},
+                },
+            )
```
Comment (jacobtylerwalls) [inline django/conf/__init__.py:232]:
```suggestion
            self.STORAGES = {
                **self.STORAGES,
                DEFAULT_STORAGE_ALIAS: {"BACKEND": self.DEFAULT_FILE_STORAGE},
            }
```

File: django/conf/__init__.py [hunk 2]
```diff
@@ -222,19 +222,29 @@ def __init__(self, settings_module):
...
-                "BACKEND": self.STATICFILES_STORAGE
-            }
+            setattr(
+                self,
+                "STORAGES",
+                {
+                    **self.STORAGES,
+                    STATICFILES_STORAGE_ALIAS: {"BACKEND": self.STATICFILES_STORAGE},
+                },
+            )
```
Comment (jacobtylerwalls) [inline django/conf/__init__.py:247]:
```suggestion
            self.STORAGES = {
                **self.STORAGES,
                STATICFILES_STORAGE_ALIAS: {"BACKEND": self.STATICFILES_STORAGE},
            }
```

**Review by jacobtylerwalls — COMMENTED** [review]
Can we get by without `setattr`?

**Comment by felixxm** [comment]
@jacobtylerwalls Thanks for checking :+1: Updated.

**Review by jacobtylerwalls — APPROVED** [review]

**Review by David-Wobrock — APPROVED** [review]
👍 

Would it make sense to add some safe guard to avoid mutating `self.STORAGES`?

**Comment by felixxm** [comment]
> 👍
> 
> Would it make sense to add some safe guard to avoid mutating `self.STORAGES`?

Thanks for checking :+1: Do you think more is needed than current tests?

**Comment by David-Wobrock** [comment]
> > 👍
> > Would it make sense to add some safe guard to avoid mutating `self.STORAGES`?
> 
> Thanks for checking 👍 Do you think more is needed than current tests?

Not necessarily. I had more in mind something in the runtime, that would raise a warning or exception every time we modify `self.STORAGES`. But that could introduce a bit of unnecessary overhead 🤔

**Pull request merged**